packages/devtools_app/lib/src/flutter/utils.dart

```diff
@@ -24,9 +24,14 @@ Future<void> copyToClipboard(List<String> lines, BuildContext context) async {
     text: lines.join('\n'),
   ));
 
-  final numLines = lines.length;
+  var numLines = lines.length;
+  // Adjust line count if there's a trailing empty line.
+  if (numLines > 0 && lines.last.trim().isEmpty) {
+    numLines--;
+  }
```

Comment:
IMO this should be fixed in the data source. Ensure that the data passed to the console never has weird empty lines (this would also allow us to remove the "render the data in the previous line" trick in the stdio stream controller)

Reply:
I think the source data is correct here - we're streaming chars into this console from stdout and stderr. There are some fiddly bits around edge cases to handle the streaming and display (and, likely some better ways we could handle those cases).

In the case I saw this issue, there were 3 lines with content, each ending in a newline, and 4 lines displayed in the console, the last one empty (w/ the user seeing 3 lines of content).

I'll double check that we shouldn't fix this in the controller - that it should just pass along 3 lines in this case.

Reply:
(updated to make changes to the controller)